compositor_render/src/transformations/web_renderer/web_renderer_thread/communication.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::Arc;
+
+use bytes::Bytes;
+use crossbeam_channel::{Receiver, Sender};
+
+use crate::{scene::ComponentId, transformations::layout::Position, Resolution};
+
+pub const EMBED_FRAMES_MESSAGE: &str = "EMBED_FRAMES";
+pub const DROP_SHARED_MEMORY: &str = "DROP_SHARED_MEMORY";
+pub const GET_FRAME_POSITIONS_MESSAGE: &str = "GET_FRAME_POSITIONS";
+
+pub struct ResponseSender<Response> {
```

Comment:
The whole point of adding this wrapper was to hide implementation details, the fact that channel are used internally, while here:
- We still need to create the channel and pass it here
- API is the same 

Using this abstraction is almost no different than just using `Option<Sender<x>>` directly. The problem that this abstraction was supposed to solve was the ability to wait for a result.

Also it does not help that you are making everything public event if its not used outside